ui/src/shared/components/DeleteDataOverlay.tsx

```diff
@@ -2,61 +2,70 @@
 import React, {FunctionComponent} from 'react'
 import {connect} from 'react-redux'
 import {withRouter, WithRouterProps} from 'react-router'
-import {Overlay} from '@influxdata/clockface'
-import {get} from 'lodash'
+import {Overlay, SpinnerContainer, TechnoSpinner} from '@influxdata/clockface'
 
 // Components
 import DeleteDataForm from 'src/shared/components/DeleteDataForm/DeleteDataForm'
 
 // Types
-import {Bucket, AppState} from 'src/types'
+import {Bucket, AppState, RemoteDataState} from 'src/types'
 
-// Utils
-import {getActiveQuery} from 'src/timeMachine/selectors'
+// Actions
+import {resetPredicateState} from 'src/shared/actions/predicates'
 
 interface StateProps {
   buckets: Bucket[]
-  selectedBucketName?: string
 }
 
-const DeleteDataOverlay: FunctionComponent<StateProps & WithRouterProps> = ({
+interface DispatchProps {
+  resetPredicateStateAction: () => void
+}
+
+type Props = WithRouterProps & DispatchProps & StateProps
+
+const DeleteDataOverlay: FunctionComponent<Props> = ({
   buckets,
   router,
   params: {orgID, bucketID},
-  selectedBucketName,
+  resetPredicateStateAction,
 }) => {
-  const handleDismiss = () =>
+  const handleDismiss = () => {
+    resetPredicateStateAction()
     router.push(`/orgs/${orgID}/load-data/buckets/${bucketID}`)
-  // separated find logic and name logic since directly routing the a delete-data
-  // endpoint was crashing the app because the bucket is undefined until the component mounts
+  }
   const bucket = buckets.find(bucket => bucket.id === bucketID)
-  const bucketName = bucket && bucket.name ? bucket.name : ''
-  const initialBucketName = selectedBucketName || bucketName
```

Comment:
how does not having a selected bucket fix the not having a selected bucket issue?

Reply:
ie: should we instead be making sure that the selected bucket is being correctly maintained throughout the application's lifecycle